File: 0137-single-number-ii/0137-single-number-ii.java
class Solution {
    public int singleNumber(int[] nums) {
        int ones = 0, two = 0;

		for(int i=0; i<nums.length; i++){
			ones = (ones ^ nums[i]) & (~two);
			two = (two ^ nums[i]) & (~ones);
		}

		return ones;
    }
}

File: 0137-single-number-ii/0137-single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        //Using Hashmap Mehthod 

        unordered_map<int , int> map;
        for(int i=0; i<nums.size(); i++){
            map[nums[i]]++;
        }

        // Iterating through the map
        for(auto it: map){
            if(it.second == 1){
                return it.first;
            }
        }
        return -1;
    }
};